### DISCUSSION TITLE
What's the reasoning behind `getLayout` in `_app.tsx`?

### DISCUSSION THREAD

**Discussion by xkrsz** [blitz-js/blitz#2866] [Q&A] [answered]
I'm wondering, what benefits does this provide over just wrapping the page component with layout directly?

**Comment by beerose** [comment]
A component might not have a layout defined, so there's a fallback to an identity function (`((page) => page)`).

**Reply by xkrsz** [reply]
Right, I get that, what I'm asking is, why does Blitz even provide this option? 🤔

**Comment by flybayer** [comment, marked as answer]
Well seems we don't have docs on this, but we should.

The reason is to achieve persisted layouts. Layouts which just wrap each page directly will be unmounted on each page transition.

See more explanation here: https://adamwathan.me/2019/10/17/persistent-layout-patterns-in-nextjs/

**Reply by xkrsz** [reply]
Ah, that makes sense! Thanks
